sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -1404,7 +1404,9 @@ case class ParseUrl(children: Seq[Expression])
     try {
       new URI(url.toString)
     } catch {
-      case e: URISyntaxException => null
+      case e: URISyntaxException if SQLConf.get.ansiEnabled =>
```

Comment:
Please update the ANSI doc, too. https://github.com/apache/spark/blame/master/docs/sql-ref-ansi-compliance.md#L110

Reply:
Missed here, updated !

Reply:
Can we follow other expressions and add a `failOnError: Boolean` parameter? See https://github.com/apache/spark/pull/30297#discussion_r521179168

Reply:
get it !